### PR TITLE
Feature canplayer

### PR DESCRIPTION
**Pull request by hardbyte**
Add framework support for replaying messages from the files we create using `can_logger.py`.
I implemented the sql db reader, and added a new script to replay with correct delays to another bus.

### CODE REVIEW

File: bin/can_player.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        datetime.datetime.now()))
+    print('Replaying data from {}'.format(
+        player.recorded_start_datetime))
+
+    try:
+        for m in in_sync:
+            bus.send(m)
+
+    except KeyboardInterrupt:
+        bus.shutdown()
```
Comment (christiansandberg) [inline bin/can_player.py]: `shutdown()` should be called also after completed execution.

File: can/CAN.py [hunk 2]
```diff
@@ -1,308 +1,379 @@
...
+        self.recorded_start_datetime = datetime.fromtimestamp(self.recorded_start_time)
+
+    @staticmethod
+    def create_frame_from_db_tuple(frame_data):
+        ts, id, is_extended, is_remote, is_error, dlc, data = frame_data
+        return Message(
+            ts, is_remote, is_extended, is_error, id, dlc, data
+        )
+
+    def __iter__(self):
```
Comment (christiansandberg) [inline can/CAN.py]: Should any open resources be closed after successful iteration, in a separate `.close()` method, or can we rely on the garbage collection?

File: can/CAN.py [hunk 3]
```diff
@@ -1,308 +1,379 @@
...
+
+class SqlReader:
+    def __init__(self, filename):
+        log.debug("Starting sqlreader with {}".format(filename))
+        conn = sqlite3.connect(filename)
+
+        self.c = conn.cursor()
+        self.c.execute("SELECT ts FROM messages LIMIT 1")
+
+        self.recorded_start_time = self.c.fetchone()[0]
```
Comment (christiansandberg) [inline can/CAN.py]: Wouldn't it be better if this was done by can_player.py instead? If it is needed at all...

Comment (hardbyte) [inline can/CAN.py]: Ahh good catch I had already moved that to the `MessageSync` in `util.py`:
```
        for m in self.raw_messages:
            if recorded_start_time is None:
                recorded_start_time = m.timestamp
```

Comment (hardbyte) [inline can/CAN.py]: I think I'll remove that message about when the trace was recorded though.

File: can/CAN.py (hunk 2 above)
Comment (hardbyte) [inline can/CAN.py]: I think that we can rely on the garbage collector - especially because we aren't writing to the db here.

**Pull request merged**